### ISSUE TITLE
X-Apollo-Tracing Headers

### ISSUE THREAD

**Issue by jasonbahl** [wp-graphql/wp-graphql-insights#16]
Tools such as graphql playground check if the GraphQL Server returns `X-Apollo-Tracing` headers to determine if the server supports tracing and if the tool should show the tracing pane. 

Something like this should be added to support tracing in external tools:

```
add_filter( 'graphql_response_headers_to_send', function( $headers ) {

	$access_control = $headers['Access-Control-Allow-Headers'];

	$headers['Access-Control-Allow-Headers'] = implode( ', ', array_merge( [
		'X-Insights-Include-Tracing',
		'X-Apollo-Tracing',
		'Credentials'
	], explode( ',', $access_control ) ) );

	return $headers;
} );
```

**Comment by CodeProKid** [comment]
@jasonbahl we've actually been having some issues with pre-flight checks & custom headers. Wondering if there should be a more centralized way for registering a header or something?

Or at lease make a central list that's filterable?

**Comment by jasonbahl** [comment]
@CodeProKid check it: https://github.com/wp-graphql/wp-graphql/pull/426

**Comment by CodeProKid** [comment]
@jasonbahl yeah I think this fulfills our needs. We just have some custom headers for checking context & such, and we were adding them ourselves, but it just felt kinda fragile being a comma separated string that we were appending the data to.
